### PR TITLE
Adding a new date formats

### PR DESCRIPTION
**Pull request by Caleydon**
Adding a new date formats to support more countries.

### CODE REVIEW

File: system/ee/EllisLab/Addons/comment/mod.comment.php [hunk 1]
```diff
@@ -952,7 +952,7 @@ function form($return_form = FALSE, $captcha = '')
 
 			// If there is no conditional- just return the message
 			ee()->lang->loadfile('comment');
-			return ee()->lang->line('cmt_commenting_has_expired');
+			return '<div class="commenting_has_expired">'.ee()->lang->line('cmt_commenting_has_expired').'</div>';
```
[the remark this answers is not in the document]
Comment (Caleydon) [reply system/ee/EllisLab/Addons/comment/mod.comment.php]: There is no way how to format this sentence when you disable comments or discussion has expired. Yes it's hard coded but adding div with class allow themes developers define CSS style and position of this sentence.

If is there better and elegant way how to achieve this goal, please fix it guys at EllisLab as a bug.

[the remark this answers is not in the document]
Comment (jmathias) [reply system/ee/EllisLab/Addons/comment/mod.comment.php]: There is a way around this output in templates, but I understand what you mean. If you install ExpressionEngine with the included default theme, you'll see how I got around this issue in `/blog/entry` (line 112+) and `/blog/_comments` (line 112).

I'm not sure if I agree that it's a bug, but if it is, I wouldn't recommend fixing it with a hardcoded HTML output. That is not something ExpressionEngine does.

If you would prefer that we treat this as a bug, you can remove it from this PR, and instead file and issue for this.

File: system/ee/EllisLab/Addons/filepicker/views/UploadView.php [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <!doctype html>
 <html>
 	<head>
-		<meta http-equiv="content-type" content="text/html; charset=utf-8" lang="en-us" dir="ltr">
+		<meta http-equiv="content-type" content="text/html; charset=utf-8" lang="<?=ee()->config->item('xml_lang')?>" dir="ltr">
```
[the remark this answers is not in the document]
Comment (Caleydon) [reply system/ee/EllisLab/Addons/filepicker/views/UploadView.php]: xml_lang is the variable defined in the exp_sites database table - site_system_preferences column. Data are ecoded by base64. 

But now I see that xml_lang option was removed  from General settings. In the previous EE generations is option available.

File: system/ee/EllisLab/ExpressionEngine/Controller/Utilities/MemberImport.php [hunk 3]
```diff
@@ -116,9 +116,14 @@ public function index()
...
-								'%j/%n/%Y' => 'dd/mm/yyyy',
-								'%j-%n-%Y' => 'dd-mm-yyyy',
+								'%n/%j/%Y' => 'm/d/yyyy',
+								'%m/%d/%Y' => 'mm/dd/yyyy',
+								'%j/%n/%Y' => 'd/m/yyyy',
+								'%d/%m/%Y' => 'dd/mm/yyyy',
+								'%j-%n-%Y' => 'd-m-yyyy',
+								'%d-%m-%Y' => 'dd-mm-yyyy',
+								'%j.%n.%Y' => 'd.m.yyyy',
+								'%d.%m.%Y' => 'dd.mm.yyyy',
```
[the remark this answers is not in the document]
Comment (Caleydon) [reply system/ee/EllisLab/ExpressionEngine/Controller/Utilities/MemberImport.php]: Please look on the original code:

```
'%n/%j/%Y' => 'mm/dd/yyyy',
'%j/%n/%Y' => 'dd/mm/yyyy',
'%j-%n-%Y' => 'dd-mm-yyyy',
```

In fact mm/dd/yyyy produce date without zero for days and months. The correct code should be:

`
'%m/%d/%Y' => 'mm/dd/yyyy',
`
or
`
'%n/%j/%Y' => 'm/d/yyyy',
`

So I included both variants for each format.

File: system/ee/EllisLab/Addons/comment/mod.comment.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (Caleydon) [reply system/ee/EllisLab/Addons/comment/mod.comment.php]: OK, ignore this commit, Please how to cancel/remove this commit?

File: system/ee/EllisLab/ExpressionEngine/Controller/Utilities/MemberImport.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (kevincupp) [reply system/ee/EllisLab/ExpressionEngine/Controller/Utilities/MemberImport.php]: This only adds the options to the Member Import screen. Why only add them here?

File: system/ee/legacy/errors/error_exception.php [hunk 4]
```diff
@@ -1,8 +1,8 @@
 <!doctype html>
 <html>
 	<head>
-		<title>Exception - ExpressionEngine</title>
-		<meta http-equiv="content-type" content="text/html; charset=utf-8" lang="en-us" dir="ltr">
+		<title>Exception | <?=APP_NAME?></title>
```
Comment (kevincupp) [inline system/ee/legacy/errors/error_exception.php]: This change seems unrelated to adding date formats.

File: system/ee/EllisLab/ExpressionEngine/Controller/Utilities/MemberImport.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (derekjones) [reply system/ee/EllisLab/ExpressionEngine/Controller/Utilities/MemberImport.php]: Can you outline what you've done to test? Formats like `m/d/yyyy` will not work, because the month and day is ambiguous, and PHP will get the date wrong frequently.

File: system/ee/legacy/errors/error_exception.php (hunk 4 above)
Comment (Caleydon) [reply system/ee/legacy/errors/error_exception.php]: Sorry, I don't know why GitHub Desktop SW merged all my pull requests into one.  Please ignore this pull request. I will create another separated.
